cinema/views.py

```diff
@@ -87,6 +100,8 @@
         )
     )
     serializer_class = MovieSessionSerializer
+    authentication_classes = (TokenAuthentication, )
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly, )
```

Comment:
The `authentication_classes` and `permission_classes` are consistently applied across all viewsets, which is good practice. Ensure that the logic in `IsAdminOrIfAuthenticatedReadOnly` aligns with the intended access control for each viewset.